File: test/integration-tests/SDK-test-suite's/javascript/functionProvider.js
//##############################################################################
// Copyright (c) 2021 Bosch.IO GmbH
//
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at https://mozilla.org/MPL/2.0/.
//
// SPDX-License-Identifier: MPL-2.0
//##############################################################################
// uncomment for local developer setup
// const iotea = require('../../../../src/module.js');
const iotea = require('boschio.iotea');

const {
    FunctionTalent,
    ProtocolGateway
} = iotea;

const {
    Logger,
    MqttProtocolAdapter,
    JsonModel
} = iotea.util;

const config = new JsonModel(require('../../config/tests/javascript/config.json'));
process.env.LOG_LEVEL = config.get('loglevel', Logger.ENV_LOG_LEVEL.INFO);

class FunctionProvider extends FunctionTalent {
    constructor(name, protocolGatewayConfig) {
        super('function-provider-js', protocolGatewayConfig);

        // Register Functions
        this.registerFunction('echo', this.echo.bind(this));
    }

    async echo(value, ev, evtctx, timeoutAtMs) {
        this.logger.debug('Echo called');
        return value;
    }
}

// TODO: make this local vs container setup configurable with ifdef
pg_config = config.get('protocolGateway');
const fp = new FunctionProvider(pg_config);

//const runner = new TestSetSDK(ProtocolGateway.createDefaultConfiguration([ MqttProtocolAdapter.createDefaultConfiguration(false,"mqtt://localhost:1883") ]));

fp.start();

File: test/integration-tests/SDK-test-suite's/python/function_provider.py
##############################################################################
# Copyright (c) 2021 Bosch.IO GmbH
#
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at https://mozilla.org/MPL/2.0/.
#
# SPDX-License-Identifier: MPL-2.0
##############################################################################
import asyncio
import logging

# Needed for protocol Gateway configuration file
# TODO: add dynamic path for directory
# import os
import json

from iotea.core.protocol_gateway import ProtocolGateway
from iotea.core.util.mqtt_client import MqttProtocolAdapter

from iotea.core.talent_func import FunctionTalent
from iotea.core.util.logger import Logger

logging.setLoggerClass(Logger)
logging.getLogger().setLevel(logging.INFO)

class FunctionProvider(FunctionTalent):
    def __init__(self, pg_config):
        super().__init__('function-provider-py', pg_config)

        # Register Functions
        self.register_function('echo', self.echo)

    # pylint: disable=invalid-name,unused-argument
    async def echo(self, value, ev, evtctx, timeout_at_ms):
        self.logger.debug('Echo called')
        return value

def read_config(abs_path):
    with open(abs_path, mode='r', encoding='utf-8') as config_file:
        return json.loads(config_file.read())

async def main():
    # TODO: add dynamic path for directory
    pg_config = read_config('../../config/tests/python/config.json')

    function_provider = FunctionProvider(pg_config['protocolGateway'])
    await function_provider.start()

if __name__ == '__main__':
    LOOP = asyncio.get_event_loop()
    LOOP.run_until_complete(main())
    LOOP.close()
